src/lib/UserSetting/UserMode.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\AdminUi\UserSetting;
+
+use Ibexa\AdminUi\Form\Type\User\UserModeChoiceType;
+use Ibexa\Contracts\Core\SiteAccess\ConfigResolverInterface;
+use Ibexa\Contracts\User\UserSetting\FormMapperInterface;
+use Ibexa\Contracts\User\UserSetting\ValueDefinitionInterface;
+use JMS\TranslationBundle\Model\Message;
+use JMS\TranslationBundle\Translation\TranslationContainerInterface;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+final class UserMode implements ValueDefinitionInterface, FormMapperInterface, TranslationContainerInterface
+{
+    public const EXPERT = '0';
+    public const SMART = '1';
+    private const TRANSLATION_DOMAIN = 'ibexa_user_settings';
+
+    private TranslatorInterface $translator;
+
+    private ConfigResolverInterface $configResolver;
+
+    public function __construct(
+        ConfigResolverInterface $configResolver,
+        TranslatorInterface $translator
+    ) {
+        $this->configResolver = $configResolver;
+        $this->translator = $translator;
+    }
+
+    public function getName(): string
+    {
+        return $this->translator->trans(
+            'user.setting.mode.name',
+            [],
+            self::TRANSLATION_DOMAIN
+        );
+    }
+
+    public function getDescription(): string
+    {
+        return $this->translator->trans(
+            'user.setting.mode.description',
+            [],
+            self::TRANSLATION_DOMAIN
+        );
+    }
+
+    public function getDisplayValue(string $storageValue): string
+    {
+        switch (true) {
+            case $storageValue === self::EXPERT:
+                return $this->translator->trans('user.setting.mode.expert', [], self::TRANSLATION_DOMAIN);
+            case $storageValue === self::SMART:
+                return $this->translator->trans('user.setting.mode.smart', [], self::TRANSLATION_DOMAIN);
+        }
+
+        return $storageValue;
+    }
+
+    public function getDefaultValue(): string
+    {
+        return $this->configResolver->getParameter('default.user_mode');
```

Comment:
This is not going to work. You need to define semantic configuration and map it to parameters first. Reference: https://github.com/ibexa/admin-ui/tree/main/src/bundle/DependencyInjection/Configuration/Parser